R/deployments.R

```diff
@@ -98,7 +98,7 @@ saveDeployment <- function(recordDir,
                            addToHistory = TRUE) {
   deployment <- deploymentRecord(
     name = deployment$name,
-    title = deployment$title,
+    title = application$title %||% deployment$title,
```

Comment:
This helps make sure that a server-updated title is reflected on the client.

Reply:
We may want to take this change even if we decide to abandon the other name-generation aspects of this PR.

Reply:
How will automatically generating a title interact with this? Will auto-generated titles overwrite titles that have been changed on Connect?